product_available_by_warehouse/views/sale_view.xml

```diff
@@ -7,7 +7,7 @@
         <field name="model">sale.order</field>
         <field name="inherit_id" ref="sale_stock.view_order_form_inherit" />
         <field name="arch" type="xml">
-            <xpath expr="//field[@name='order_line']/form//field[@name='product_id']" position="before">
+            <xpath expr="//field[@name='order_line']/form//field[@name='product_id']" position="after">
```

Comment:
@nhomar 
in fact, it is requirement for the customer, in usability is better to choose first the product
![produtc_after](https://cloud.githubusercontent.com/assets/7595505/18501578/393abfc6-7a17-11e6-8dad-efaca7c25ac9.png)
 than this
![produt_before](https://cloud.githubusercontent.com/assets/7595505/18501595/5279363e-7a17-11e6-9aa4-18229491baf7.png)